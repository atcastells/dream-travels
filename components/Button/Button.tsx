import classNames from "classnames";
import {
  ButtonHTMLAttributes,
  DetailedHTMLProps,
  PropsWithChildren,
} from "react";

export type ButtonProps = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
> &
  PropsWithChildren<{}>;
export const Button = ({ children, ...props }: ButtonProps) => {
  return (
    <button
      {...props}
      className={classNames(
        props?.className ? props.className : "",
        "rounded-full px-2"
      )}
    >
      {children}
    </button>
  );
};
