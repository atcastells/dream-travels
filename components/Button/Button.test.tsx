import { expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";

import { Button } from "./Button";

test("Button renders children", () => {
  render(<Button className="text-red-400">Click me</Button>);
  expect(screen.getByText("Click me")).toBeTruthy();

  const button = screen.getByRole("button");

  expect(button.className).toContain("text-red-400");

  //Snapshot testing
  expect(button).toMatchSnapshot();
});
