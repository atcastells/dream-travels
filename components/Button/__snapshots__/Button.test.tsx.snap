// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders children 1`] = `
<button
  class="text-red-400 rounded-full px-2"
>
  Click me
</button>
`;
